From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Mon, 24 Jan 2011 19:26:48 +0000 (-0600)
Subject: ASoC: ABE DSP: Set aess clock rate before loading firmware
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=9dbe0275e6a07a783cf679b34061c68ad7a24ebd;hp=9a0ec11b2f9b9e853c8808779e5f6aed0de46960

ASoC: ABE DSP: Set aess clock rate before loading firmware

ABE firmware needs to be reloaded after ABE domain hits to OFF,
but it's required to set aess clock rate before firmware is
reloaded to ensure proper operation of ABE.

Change-Id: If30fd8042723884cc4ddcea512662469e4d13378
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
---

Index: android-2.6.35/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- android-2.6.35.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-25 11:47:33.147811916 +0530
+++ android-2.6.35/sound/soc/omap/omap-abe-dsp.c	2011-01-25 11:54:02.277832749 +0530
@@ -1764,6 +1764,8 @@
 	struct omap4_abe_dsp_pdata *pdata = pdev->dev.platform_data;
 	int loss_count = 0;
 
+	omap_device_set_rate(&pdev->dev, &pdev->dev, 98000000);
+
 	if (pdata->get_context_loss_count)
 		loss_count = pdata->get_context_loss_count(&pdev->dev);
 
